src/baal/utils/pytorch_lightning.py

```diff
@@ -70,3 +98,38 @@ def predict_on_dataset_generator(self, *args, **kwargs):
             yield map_on_tensor(lambda x: x.detach().cpu().numpy(), pred)
         # teardown, TODO customize this later?
         model.cpu()
+
+    def step(self, pool=None) -> bool:
+        """
+        Perform an active learning step.
+
+        Args:
+            pool (iterable): dataset pool indices.
+
+        Returns:
+            boolean, Flag indicating if we continue training.
+
+        """
+        # High to low
+        if pool is None:
```

Comment:
In this case we can remove everything and just call self.predict_on_dataset_generator(**self.kwargs) I think? We will call the pool_loader.